pymc_extras/statespace/core/statespace.py

```diff
@@ -559,14 +567,17 @@ def make_and_register_data(
                 f"parameters."
             )
 
-        if name in self._name_to_data.keys():
+        if name in self._tensor_data_info:
             raise ValueError(
                 f"{name} is already a registered placeholder variable with shape "
-                f"{self._name_to_data[name].type.shape}"
+                f"{self._tensor_data_info[name].type.shape}"
```

Comment:
Same as above, use `add` so the duplicate check is being done by the `Info`. The `StateSpace` class itself doesn't need to know about that logic.